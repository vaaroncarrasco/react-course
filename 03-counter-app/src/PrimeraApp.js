import React from "react";
import PropTypes from 'prop-types';

// ? Functional Components
const PrimeraApp = ({ saludo, subtitulo }) => {
    // * Return elements in Component -> Fragment <></>
    // <></> is the same as <Fragment></Fragment>
    // return () encapsulates one object/fragment/element in one return
    return (
        <>
            {/* printing objects */}
            {/* <pre> { JSON.stringify(object, null, 3) } </pre> */}
            <h1>{ saludo }</h1>
            <p>{ subtitulo }</p>
        </>
    );

    // return (
    //     <Fragment>
    //         <h1>Hola mundo</h1>
    //         <p>Mi primera app</p>
    //     </Fragment>
    // );
}

// Component properties expected types
PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired
}

PrimeraApp.defaultProps = {
    subtitulo: 'soy un texto de prueba'
}

export default PrimeraApp;