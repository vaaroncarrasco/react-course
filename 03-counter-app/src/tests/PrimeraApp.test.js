import React from "react";
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import PrimeraApp from "../PrimeraApp";

describe('<PrimeraApp /> testing', () => {

    // test('should show the msg "hola mundo"', () => {
    //     const saludo = 'hola mundo';
    //     // wrapper is a library that renders react components to test them
    //     // const wrapper = render(<PrimeraApp saludo={ saludo } />);
    //     const { getByText } = render(<PrimeraApp saludo={ saludo } />);
    //     expect( getByText(saludo) ).toBeInTheDocument();
    // });

    // * Install enzyme then the adapter
    test('should show <PrimeraApp /> correctly', () => {
        const saludo = 'hola mundo';
        const wrapper = shallow( <PrimeraApp saludo={ saludo } /> ); // shallow comes with enzyme package

        // * enzyme snapshot to json -> for testing purposes
        expect( wrapper ).toMatchSnapshot(); // to match snapshot from original react component
    });

    test('should show subtitle sent through props', () => {
        const saludo = 'hola mundo';
        const subtitulo = 'texto de prueba';

        // * wrapper has all the data of the rendered component
        const wrapper = shallow( <PrimeraApp saludo={ saludo } subtitulo={ subtitulo } />);

        const textoParrafo = wrapper.find('p').text();

        expect( textoParrafo ).toBe( subtitulo );
    });
});
