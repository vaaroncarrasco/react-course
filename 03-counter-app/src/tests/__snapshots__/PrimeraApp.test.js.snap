// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrimeraApp /> testing should show <PrimeraApp /> correctly 1`] = `
<Fragment>
  <h1>
    hola mundo
  </h1>
  <p>
    soy un texto de prueba
  </p>
</Fragment>
`;
